### PR TITLE
Expose coreboot version information in coreboot table for consumption by EDK

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
New CONFIG_DRIVERS_EFI_FW_INFO is off by default, enabling it adds DRIVERS_EFI_FW_{GUID,VERSION,LSV} to be used to specify firmware version/update information.

Existing forms of versions woudn't be sufficient because there is no universal way of conversion to 32-bit unsigned integers and there are no GUIDs or lowest supported versions.

***

Related to https://github.com/Dasharo/coreboot/pull/509, but this functionality can be enabled independently so I didn't tie them together.

EDK part: https://github.com/Dasharo/edk2/pull/145

### CODE REVIEW

File: src/commonlib/include/commonlib/coreboot_tables.h [hunk 1]
```diff
@@ -87,6 +87,7 @@ enum {
 	LB_TAG_TYPE_C_INFO		= 0x0042,
 	LB_TAG_ACPI_RSDP		= 0x0043,
 	LB_TAG_PCIE			= 0x0044,
+	LB_TAG_EFI_FW_INFO		= 0x0046,
```
Comment (miczyg1) [inline src/commonlib/include/commonlib/coreboot_tables.h:90]: Is 0x45 occupied already in upstream?

Comment (krystian-hebel) [reply src/commonlib/include/commonlib/coreboot_tables.h:90]: No, but it is by capsule data in #509

**Review by miczyg1 — DISMISSED** [review]

File: src/lib/coreboot_table.c [hunk 2]
```diff
@@ -572,6 +572,10 @@ static uintptr_t write_coreboot_table(uintptr_t rom_table_end)
 	/* Add board-specific table entries, if any. */
 	lb_board(head);
 
+	/* Add information about firmware in form suitable for EFI updates. */
```
Comment (krystian-hebel) [inline src/lib/coreboot_table.c:575]: Maybe do this before `lb_board()`? That function is most likely the place where other components will be added, it seems logical to have main FW as first entry.

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

File: src/lib/coreboot_table.c (hunk 2 above)
Comment (SergiiDmytruk) [reply src/lib/coreboot_table.c:572]: Moved: https://github.com/Dasharo/coreboot/compare/00bebe0e391889f56d2891b310d7a7250680c021..702c924defd5047017c5ea5f154e41f0d96b7d02#diff-ceee6c791f39074f69b3cc196251618d0d22a8c3c7227f6b44783c08ec69b51f

File: src/include/boot/coreboot_tables.h [hunk 3]
```diff
@@ -27,6 +27,9 @@ enum cb_err fill_lb_pcie(struct lb_pcie *pcie);
 /* Define this in mainboard.c to add board-specific table entries. */
 void lb_board(struct lb_header *header);
 
+/* Adds LB_TAG_EFI_FW_INFO table entry (or entries). */
+void lb_efi_fw_info(struct lb_header *header);
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply src/include/boot/coreboot_tables.h]: I don't really know where other components will be added, so dropped parenthesized part: https://github.com/Dasharo/coreboot/compare/00bebe0e391889f56d2891b310d7a7250680c021..702c924defd5047017c5ea5f154e41f0d96b7d02#diff-b725a4c20f8592a04953600959de75fd2465eb9011572328677f531f047d524f

Given that GUIDs will differ between boards, maybe some kind of component type is necessary for EDK to be able to know how to perform flashing of a specific ESRT entry.

File: src/drivers/efi/Kconfig [hunk 4]
```diff
@@ -6,3 +6,35 @@ config DRIVERS_EFI_VARIABLE_STORE
...
 	  VariableStore as used by tianocore.
+
+config DRIVERS_EFI_FW_INFO
+	bool "Expose firmware version in a EFI-friendly form"
+	depends on UDK_BASE
+	help
+	  Adds firmware version information to coreboot table in a form similar to
+	  EFI System Resource Table (ESRT) that can be used for firmware updates.
+
+config DRIVERS_EFI_FW_GUID
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply src/drivers/efi/Kconfig]: Added `MAIN_`: https://github.com/Dasharo/coreboot/compare/00bebe0e391889f56d2891b310d7a7250680c021..702c924defd5047017c5ea5f154e41f0d96b7d02#diff-7ffa6040ba95cc7406e87024010ef25159653dc594f56d00205c9fe95825f3ff

**Review by krystian-hebel — APPROVED** [review]
I'm approving these changes, but to keep things clean in case of further issues discovered when working on next tasks, it would be best to keep it on separate branch until we have fully working capsule updates. This would also help keeping the changes in sync with upstreamed versions.

**Comment by krystian-hebel** [comment]
Closing this PR, changes are in #509 (it has more fitting branch name) and work will be continued there.

**Pull request closed without merging**